vm/devices/support/fs/lxutil/src/windows/symlink.rs

```diff
@@ -2,15 +2,15 @@
 // Licensed under the MIT License.
 
 use pal::windows::UnicodeString;
-use pal::windows::UnicodeStringRef;
 use windows::Wdk::Storage::FileSystem;
 use windows::Win32::System::SystemServices as W32Ss;
 
 /// Get the symlink substitute name and flags from the reparse data.
-fn get_substitute_name(
+/// The caller must guarantee that the reparse data buffer is well-formed.
+unsafe fn get_substitute_name(
```

Comment:
I think we can do better in handling all this reparse stuff. We really want to lean on Rust more to help find memory safety bugs, instead of just adding `unsafe`. The caller has a `HeaderVec` that knows the full size of the structure, but we lose that information as part of passing it down here.

This is OK for now, but add a TODO to eliminate the need for unsafe here.